src/extension.ts

```diff
@@ -31,8 +34,10 @@ export async function activate(context: vscode.ExtensionContext) {
   var authService = new AuthenticationService(context, telemetryService);
 
   // Setup our CodeParser service
-  const pending = new CodeParserService(context);
-  var codeParserService = await pending;
+  var codeParserService = await createCodeParserService(
+    context,
+    telemetryService
+  );
```

Comment:
Again, not sure if I'm a fan of having the service contain a function to create itself? Why not use a constructor as with other services?

Reply:
This makes it so we can treat the service as a singleton. The issue we were having with tests was that multiple code parser services were trying to be created, and they were all trying to register commands. This way there will only ever be one instance of the service, and we can fetch it using the getter method

Reply:
I think service might just be the wrong name here - we aren't really making a service, more an instance of a class where a singleton makes sense. Either way, I'm much more comfortable doing this here with the code parser than I am with the telemetry service.